data/src/main/java/com/microsoft/azure/kusto/data/ClientRequestProperties.java

```diff
@@ -70,25 +77,51 @@ public Long getTimeoutInMilliSec() {
         if (timeoutObj instanceof Long) {
             timeout = (Long) timeoutObj;
         } else if (timeoutObj instanceof String) {
-            timeout = LocalTime.parse((String) timeoutObj).toNanoOfDay() / NANOS_TO_MILLIS;
+            timeout = getMillisFromTimespanString((String) timeoutObj);
         } else if (timeoutObj instanceof Integer) {
             timeout = Long.valueOf((Integer) timeoutObj);
         }
 
         return timeout;
     }
 
+    private long getMillisFromTimespanString(String str) throws TimespanParseException {
+        Matcher matcher = PATTERN.matcher(str);
+        if (!matcher.matches()) {
+            throw new TimespanParseException(str);
+        }
+
+        long millis = 0;
+        String days = matcher.group(1);
+        if (days != null) {
+            millis += Long.parseLong(days) * SECONDS_PER_DAY * 1000L;
```

Comment:
As above, I think Java's Duration class is better suited instead of reinventing the wheel.

Java's time library also already has all these constants covered. For example, `SECONDS_PER_DAY` would be:
`long seconds = TimeUnit.DAYS.toSeconds(1)`
And the millis you're looking for would be:
`long millis = TimeUnit.DAYS.toMillis(1)`